xray/segment.go

```diff
@@ -83,8 +83,8 @@ func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request,
 	}
 
 	if r == nil || traceHeader == nil {
-		// Sampling strategy fallbacks to default sampling in the case of directly using BeginSegment API without any instrumentation
-		sd := seg.ParentSegment.GetConfiguration().SamplingStrategy.ShouldTrace(&sampling.Request{})
+		// No header or request information provided so we can only evaluate sampling based on the serviceName
```

Comment:
Also, can we include this as a Note in README file so that sampling behavior is very clear when customer use BeginSegment API directly. (since we are evaluating only based on service name here)